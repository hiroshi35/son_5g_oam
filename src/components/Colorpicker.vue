<template>
  <div>
    <div v-if="icon" class='input-group' ref="element">
      <input type='text' class="form-control" :placeholder="placeholder" :value="value">
      <span class="input-group-addon"><i></i></span>
    </div>
    <input v-else type="text" class="form-control" :placeholder="placeholder" ref="element" :value="value">
  </div>

</template>
<script>
  import 'bootstrap-colorpicker'
  export default {
    props: {
      placeholder: {
        type: String,
        default: 'Enter Color'
      },
      options: {
        type: Object,
        default () {
          return {
            format: 'hex'
          }
        }
      },
      icon: {
        type: Boolean,
        default: false
      },
      value: {
        type: String
      }
    },
    watch: {
      color (val) {
        this.$emit('input', val)
      },
      value (val) {
        $(this.$refs.element).colorpicker('setValue', val)
      }
    },
    mounted () {
      let vm = this
      $(this.$refs.element).colorpicker(vm.options)
      $(this.$refs.element).colorpicker().on('changeColor', function (e) {
        vm.color = e.color.toHex()
      })
    },
    data () {
      return {
        color: ''
      }
    }
  }
</script>

<style lang="scss">
  @import "~bootstrap-colorpicker/dist/css/bootstrap-colorpicker";
</style>

