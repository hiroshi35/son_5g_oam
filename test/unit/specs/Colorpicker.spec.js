import { shallow } from 'vue-test-utils'
import Colorpicker from '@/components/Colorpicker.vue'
import {colorpicker, on} from 'jquery'
describe('Colorpicker.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Colorpicker)
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('renders .input-group-addon', () => {
    wrapper.setProps({icon: true})
    expect(wrapper.find('.input-group-addon').is('span')).toMatchSnapshot()
  })
  test('$().colorpicker() should be called', () => {
    expect(colorpicker).toBeCalled()
  })
  test('$().colorpicker().on() should be called with correct argument', () => {
    expect(on).toBeCalled()
    expect(on.mock.calls[0][0]).toBe('changeColor')
  })
  test('event should be emitted when color is changed', () => {
    wrapper.setData({color: '#fff'})
    expect(wrapper.emitted().input[0][0]).toBe('#fff')
  })
})
