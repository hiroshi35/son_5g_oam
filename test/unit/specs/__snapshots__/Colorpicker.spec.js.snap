// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colorpicker.vue has the expected html structure 1`] = `
<div>
  <input
    class="form-control"
    placeholder="Enter Color"
    type="text"
  />
</div>
`;

exports[`Colorpicker.vue renders .input-group-addon 1`] = `true`;
